tests/post_training/test_templates/test_batch_size.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC
+from abc import abstractmethod
+from dataclasses import dataclass
+from dataclasses import fields
+
+import pytest
+
+from nncf.common.tensor_statistics.aggregator import StatisticsAggregator
+from tests.post_training.test_templates.helpers import get_static_dataset
+
+
+@dataclass
+class DataForTest:
+    batch_size: int
+    dataset_len: int
+    stat_subset_size: int
+    ref_calibration_samples_num: int
+    ref_iterations_num: int
+
+
+class TemplateTestBatchSize(ABC):
+    @abstractmethod
+    def create_statistics_aggregator(self, dataset) -> StatisticsAggregator:
+        pass
+
+    def create_dataset(self, lenght, batch_size):
+        dataset = get_static_dataset(None, None, None, lenght)
+        dataset._data_source.batch_size = batch_size
+        return dataset
+
+    @pytest.mark.parametrize(
+        ("test_data"),
+        (
+            [  # batch_size | dataset_len | stat_subset_size | ref_calibration_samples_num | ref_iterations_num
+                # DataForTest(None, None, None, None, None),  # None is None
```

Comment:
The commented-out line for DataForTest(None, None, None, None, None) is unnecessary and could be removed to clean up the code.

Consider removing the commented-out line to keep the codebase clean and focused on active test cases.

Reply:
Same